public-ui/src/app/views/ehr-view/ehr-view.component.html

```diff
@@ -65,33 +65,61 @@ <h5 class="col-xs-12 box-title">
     <div class="white-card-box">
 
       <div class="results-grid" *ngIf="items.length > 0" >
-        <div class="row" *ngIf="items.length === 1 && items[0].standardConcept != 'S'">
+        <div class="row" *ngIf="items.length === 1 && items[0].standardConcept != 'S' && standardConcepts.length > 0">
           <div class="col-xs-12 db-alert">
             <clr-alert  [clrAlertClosable]="false" [clrAlertType]="info">
               Note: {{items[0].vocabularyId}} {{items[0].conceptCode}} "{{items[0].conceptName}}"
-              maps to a Standard Vocabulary {{standardConcepts[0].vocabularyId}} {{standardConcepts[0].conceptCode}}
-              and any listed below. Standard vocabularies capture data across a variety of source vocabularies.
+              maps to Standard Vocabulary {{standardConcepts[0].vocabularyId}} {{standardConcepts[0].conceptCode}}.
+              Standard vocabularies capture data across a variety of source vocabularies.
             </clr-alert>
           </div>
         </div>
         <div class="row">
-          <div class="col-md grid-header flex-xs-middle"> {{dbDomain.domainDisplay}}</div>
-          <div class="col-md grid-header flex-xs-middle"> Synonyms </div>
-          <div class="col-md-2 grid-header flex-xs-middle"> Participants of <b> {{totalParticipants | number}}</b> </div>
+          <div class="col-md-3 grid-header flex-xs-middle"> Concept Name</div>
+            <div class="col-md-4 grid-header flex-xs-middle"> Synonyms </div>
+          <div class="col-md-2 grid-header flex-items-xs-right"> Participants of <b> {{totalParticipants | number}}</b> </div>
           <div class="col-md-2 grid-header flex-xs-middle info-text">
             Vocabulary Code
             <clr-icon class="" shape="info-standard" class="is-solid"></clr-icon>
           </div>
           <div class="col-md-1 grid-header flex-xs-middle info-text"> View graphs </div>
-
         </div>
         <div class="row expandable-grid-row"
-             *ngFor="let r of items.concat(standardConcepts), let i = index"
+             *ngFor="let r of standardConcepts"
              [ngClass]="{'expanded-grid-row': r.expanded}" >
-          <div class="col-md flex-xs-middle" innerHtml="{{i+1}}. {{r.conceptName | highlightSearch : prevSearchText}}
+          <div class="col-md-3 flex-xs-middle" innerHtml=" <b>Standard</b> {{r.conceptName | highlightSearch : prevSearchText}}
+            -- {{r.conceptId}}"> </div>
+          <div class="col-md-4 flex-xs-middle" *ngIf="r.conceptSynonyms;else noConceptSynonyms" innerHtml="{{ r.conceptSynonyms.join(', ') | highlightSearch : prevSearchText}}"></div>
```

Comment:
Why the ngIf ? Does this give an error if there are no synonyms ?  

Also could you add a custom pipe like this to truncate the synonyms to the length we want ? I think that will be a good safeguard to have in place.

Reply:
Yeah, it was throwing error before on using join on undefined.